File: src/main/java/org/asm/immomanage/dto/propertyDto/PropertyRequestDto.java
package org.asm.immomanage.dto.propertyDto;

import lombok.Builder;
import org.asm.immomanage.dto.propertyEquipmentsDto.PropertyEquipmentDto;
import org.asm.immomanage.models.ImageModel;
import org.asm.immomanage.utils.Status;

import java.util.Set;

@Builder
public record PropertyRequestDto(
        double rentPrice,
        String description,
        String address,
        Status status,
        Set<PropertyEquipmentDto> propertyEquipmentDto,
        Set<ImageModel> propertyImages,
        Set<Long> tenantsIDS,
        Long managerID
) {
}

File: ImmoMange-ui/src/app/services/models/property-request-dto.ts
/* tslint:disable */
/* eslint-disable */
import { PropertyEquipmentDto } from '../models/property-equipment-dto';
import { PropertyImageDto } from '../models/property-image-dto';
export interface PropertyRequestDto {
  address?: string;
  cinTenants?: Array<string>;
  description?: string;
  managerEmail?: string;
  propertyEquipmentDto?: Array<PropertyEquipmentDto>;
  propertyImages?: Array<PropertyImageDto>;
  rentPrice?: number;
  status?: 'OCCUPIED' | 'AVAILABLE' | 'UNDER_MAINTENANCE';
}
